app/models/participant_declaration.rb

```diff
@@ -89,7 +92,26 @@ def make_paid!
     DeclarationState.paid!(self) if payable?
   end
 
+  def make_ineligible!(reason: nil)
+    DeclarationState.ineligible!(self, state_reason: reason) if submitted?
+  end
+
   def changeable?
     %w[submitted eligible].include?(current_state)
   end
+
+  def duplicate_declarations
+    self.class.joins(participant_profile: :teacher_profile)
+      .where(participant_profiles: { teacher_profiles: { trn: participant_profile.teacher_profile.trn } })
+      .where.not(participant_profiles: { teacher_profiles: { trn: nil } })
+      .where.not(user_id: user_id, id: id)
+      .where(
+        declaration_type: declaration_type,
+        course_identifier: course_identifier,
+        superseded_by_id: nil,
```

Comment:
I think we'll want `voided: false` here as well — we should allow them to post dups of voided declarations.